File: HAS-A.java
//Implementing Has-A Relationship in Java


class Shape {
    String type;
    public Shape(String type) {
        this.type = type;
    }
    void draw() {
        System.out.println("Drawing a " + type);
    }
}
class Drawing {
    Shape drawingShape;
    public Drawing(Shape drawingShape) {
        this.drawingShape = drawingShape;
    }
    void drawPicture() {
        System.out.println("Creating a picture by drawing:");
        drawingShape.draw();
    }
}
public class Main {
    public static void main(String[] args) {
        Shape myShape = new Shape("Circle");
        Drawing myDrawing = new Drawing(myShape);
        myDrawing.drawPicture();
    }
}

File: HAS-A.js
//Implementing Has-A Relationship in JavaScript


class Shape {
    constructor(type) {
        this.type = type;
    }
    draw() {
        console.log("Drawing a " + this.type);
    }
}
class Drawing {
    constructor(drawingShape) {
        this.drawingShape = drawingShape;
    }
    drawPicture() {
        console.log("Creating a picture by drawing:");
        this.drawingShape.draw();
    }
}
const myShape = new Shape("Circle");
const myDrawing = new Drawing(myShape);
myDrawing.drawPicture();
